src/compound/lens/IndexesLens.sol

```diff
@@ -115,44 +93,122 @@ abstract contract IndexesLens is LensStorage {
             indexes.poolSupplyIndex = ICToken(_poolToken).exchangeRateStored();
             indexes.poolBorrowIndex = ICToken(_poolToken).borrowIndex();
         } else {
-            (indexes.poolSupplyIndex, indexes.poolBorrowIndex) = getCurrentPoolIndexes(_poolToken);
+            (indexes.poolSupplyIndex, indexes.poolBorrowIndex, ) = _accruePoolInterests(
+                ICToken(_poolToken)
+            );
         }
 
-        if (!_updated || block.number == lastPoolIndexes.lastUpdateBlockNumber) {
-            indexes.p2pSupplyIndex = morpho.p2pSupplyIndex(_poolToken);
-            indexes.p2pBorrowIndex = morpho.p2pBorrowIndex(_poolToken);
+        (indexes.p2pSupplyIndex, indexes.p2pBorrowIndex, ) = _computeP2PIndexes(
+            _poolToken,
+            _updated,
+            indexes.poolSupplyIndex,
+            indexes.poolBorrowIndex,
+            delta,
+            lastPoolIndexes
+        );
+    }
+
+    /// @notice Returns the virtually updated pool indexes of a given market.
+    /// @dev Mimicks `CToken.accrueInterest`'s calculations, without writing to the storage.
+    /// @param _poolToken The address of the market.
+    /// @return poolSupplyIndex The supply index.
+    /// @return poolBorrowIndex The borrow index.
+    function _accruePoolInterests(ICToken _poolToken)
+        internal
+        view
+        returns (
+            uint256 poolSupplyIndex,
+            uint256 poolBorrowIndex,
+            PoolInterestsVars memory vars
+        )
+    {
+        poolBorrowIndex = _poolToken.borrowIndex();
```

Comment:
I find it kind of misleading and sometime kind of dangerous to use returned variable for something else (in the case where `block.number != accrualBlockNumberPrior` only). I would rather use an explicit return in the first if block.

```suggestion
```

Reply:
Why is it dangerous? The variable is assigned before early returning

Reply:
Because the variable is assigned to an other value than what it should be at the end.

Reply:
But here it's not? Whatever the value of `block.number != accrualBlockNumberPrior`, the value should be the same and equal to `_poolToken.borrowIndex()`?

Reply:
But at the end you change the value right 

```solidity
poolBorrowIndex += simpleInterestFactor.mul(poolBorrowIndex);
```

Reply:
Ok, so my argument was wrong

In this case, you are against having intermediate values for variables then?
Because in many cases it is useful to store some intermediate value to a variable (either because it's a correct value at time of execution of the assignment, or because it saves stack/memory/gas)